javascript/node/selenium-webdriver/lib/webdriver.js

```diff
@@ -1515,6 +1520,110 @@ class WebDriver {
       }
     })
   }
+
+  /**
+   * 
+   * @returns The value of authenticator ID added
+   */
+  virtualAuthenticatorId() {
+    return this.authenticatorId_
+  }
+
+  /**
+   * Adds a virtual authenticator with the given options.
+   * @param options VirtualAuthenticatorOptions object to set authenticator optons.
+   */
+  async addVirtualAuthenticator(options) {
+    this.authenticatorId_ = await this.execute(
+      new command.Command(command.Name.ADD_VIRTUAL_AUTHENTICATOR).setParameters(
+        options.toDict()
+      )
+    )
+  }
+
+  /**
+   * Removes a previously added virtual authenticator. The authenticator is no
+   * longer valid after removal, so no methods may be called.
+   */
+  async removeVirtualAuthenticator() {
+    await this.execute(
+      new command.Command(
+        command.Name.REMOVE_VIRTUAL_AUTHENTICATOR
+      ).setParameter('authenticatorId', this.authenticatorId_)
+    )
+    this.authenticatorId_ = null
```

Comment:
What if the user wants to register multiple authenticators?

Reply:
Right. I'll fix that.

Reply:
The global variable led to that thought. The question is more of will there ever be such a situation. It does not intend to question the current implementation. I think we can let it be how it is. I thought about it and I am unable to foresee a situation that needs multiple authenticators. We can tackle it if and when the need arises.